src/chocolatey/infrastructure.app/services/ISourceRunner.cs

```diff
@@ -25,122 +25,164 @@ namespace chocolatey.infrastructure.app.services
     using results;
 
     [MultiService]
-    public interface ISourceRunner
+    [Obsolete("This interface is deprecated and will be removed in v3.")]
+    public interface ISourceRunner : IBootstrappableSourceRunner, ICountSourceRunner, IListSourceRunner, IInstallSourceRunner, IUpgradeSourceRunner, IUninstallSourceRunner
+    {
+#pragma warning disable IDE1006
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        void ensure_source_app_installed(ChocolateyConfiguration config, Action<PackageResult, ChocolateyConfiguration> ensureAction);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        int count_run(ChocolateyConfiguration config);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        void list_noop(ChocolateyConfiguration config);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        IEnumerable<PackageResult> list_run(ChocolateyConfiguration config);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        void install_noop(ChocolateyConfiguration config, Action<PackageResult, ChocolateyConfiguration> continueAction);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        ConcurrentDictionary<string, PackageResult> install_run(ChocolateyConfiguration config, Action<PackageResult, ChocolateyConfiguration> continueAction);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        ConcurrentDictionary<string, PackageResult> install_run(ChocolateyConfiguration config, Action<PackageResult, ChocolateyConfiguration> continueAction, Action<PackageResult, ChocolateyConfiguration> beforeModifyAction);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        ConcurrentDictionary<string, PackageResult> upgrade_noop(ChocolateyConfiguration config, Action<PackageResult, ChocolateyConfiguration> continueAction);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        ConcurrentDictionary<string, PackageResult> upgrade_run(ChocolateyConfiguration config, Action<PackageResult, ChocolateyConfiguration> continueAction, Action<PackageResult, ChocolateyConfiguration> beforeUpgradeAction = null);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        void uninstall_noop(ChocolateyConfiguration config, Action<PackageResult, ChocolateyConfiguration> continueAction);
+        [Obsolete("This overload is deprecated and will be removed in v3.")]
+        ConcurrentDictionary<string, PackageResult> uninstall_run(ChocolateyConfiguration config, Action<PackageResult, ChocolateyConfiguration> continueAction, Action<PackageResult, ChocolateyConfiguration> beforeUninstallAction = null);
+#pragma warning restore IDE1006
+    }
+
+    public interface IAlternativeSourceRunner

```

Comment:
Also, since we're using this for all our sourcerunners, NugetService included (even if it's still using the main ISourceRunner, it gets this by virtue of the inherited interfaces) I feel like it's a bit weird to call this "Alternative".

If we specifically want a new name for the interface maybe we could go with something like `IPackageSource` or something along those lines?

Though, given the rest of the now-very-breaking changes added here, I think we could just skip the deprecation here anyway and replace ISourceRunner completely with this interface...

Reply:
The`Alternative` here refers to Alternative Sources, which is the terminology used for things like Cygwin, Python, etc.  You can see that in our docs here: https://docs.chocolatey.org/en-us/choco/commands/search#alternative-sources